Request/ParamConverter/PsrServerRequestParamConverter.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Converts HttpFoundation Request to PSR-7 ServerRequest using the bridge.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PsrServerRequestParamConverter implements ParamConverterInterface
+{
+    private static $supportedTypes = [
+        'Psr\Http\Message\ServerRequestInterface' => true,
+        'Psr\Http\Message\RequestInterface' => true,
+        'Psr\Http\Message\HttpMessageInterface' => true,
+    ];
```

Comment:
Goof catch!

The converter will work with PHP 5.3 as long a you use (create for now) a PSR-7 implementation compliant with it. The only package requiring PHP 5.5+ is Diactoros and it's a soft dependency.